iFixit/react-commerce
# Prevent Algolia hooks in product list empty state from unmounting

fixes #398 

This subtle bug is due to Algolia library behaviour. Whenever an Algolia hook or widget gets unmounted, the state is reset. I think this is not a wise API design choice because it makes it really easy to introduce subtle bugs, like in this case.
Here the problem is that the product list empty state makes use of `useSearchBox`. To fix the issue we're keeping the component mounted, but hidden via CSS when it's not needed.

## Changelog

- Fix the bug
- Remove unused dependencies
- Formatting adjustments: these are due to the fact that @federicobadini has fixed a discrepancy between prettier versions in the project, so many files are still not correctly formatted, and are being formatted when you change them.

## QA

1. Open the vercel preview
2. Verify that #398 is fixed
